sunscreen_runtime/src/builder.rs

```diff
@@ -409,44 +477,89 @@ mod linked {
             self.encrypt_internal(Message::Linked(message.clone()), public_key, bounds)
         }
 
+        /// Encrypt a linked message symmetrically, adding the new encryption statement to the
+        /// proof.
+        ///
+        /// This method purposefully reveals that two ciphertexts enrypt the same underlying value. If
+        /// this is not what you want, use [`Self::encrypt`].
+        ///
+        /// This method assumes that you've created the `message` argument with _this_ builder.
+        pub fn encrypt_symmetric_linked(
+            &mut self,
+            message: &LinkedMessage,
+            private_key: &'k PrivateKey,
+        ) -> Result<Ciphertext> {
+            // The existing message already has bounds, no need to recompute them.
+            let bounds = None;
+            self.encrypt_symmetric_internal(Message::Linked(message.clone()), private_key, bounds)
+        }
+
         fn encrypt_internal(
             &mut self,
             message: Message,
             public_key: &'k PublicKey,
             bounds: Option<Bounds>,
         ) -> Result<Ciphertext> {
-            let enc_components = self.runtime.encrypt_return_components_switched_internal(
-                message.pt(),
-                message.type_name(),
-                public_key,
-                true,
-                None,
-            )?;
             let existing_idx = message.linked_id();
+            let mut i = 0;
```

Comment:
Similarly we could consider renaming `encrypt_internal` with `encrypt_asymmetric_internal`.